File: aspnet-core/src/HRMv2.Core/Manager/Debts/Dto/DebtDto.cs
using Abp.AutoMapper;
using HRMv2.Entities;
using HRMv2.Manager.Common.Dto;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static HRMv2.Constants.Enum.HRMEnum;

namespace HRMv2.Manager.Debts.Dto
{
    [AutoMapTo(typeof(Debt))]
    public class DebtDto : BaseEmployeeDto
    {
        public long? EmployeeId { get; set; }
        public double InterestRate { get; set; }
        public double Money { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string Note { get; set; }
        public DebtPaymentType PaymentType { get; set; }
        public DebtStatus DebtStatus { get; set; }
        public double TotalPaid { get; set; }
        public DateTime? UpdatedTime { get; set; }
        public string UpdatedUser { get; set; }
        public string CreatorUser { get; set; }
        public DateTime CreationTime { get; set; }
        public string IdCard { get; set; }
        public string IssuedBy { get; set; }
        public DateTime? IssuedOn { get; set; }

    }
}

File: angular/src/app/service/model/debt/debt.dto.ts
import { BadgeInfoDto, BaseEmployeeDto } from "@shared/dto/user-infoDto";
import { GetEmployeeDto } from "../employee/employee.dto"
import { GetInputFilterDto } from "../employee/GetEmployeeExcept.dto";

export interface DebtDto {
    id: number,
    fullName: string,
    email: string,
    sex: number,
    jobPosition: string,
    jobPositionId: number,
    userType: string,
    userTypeId: number,
    team: string,
    teamId: number,
    status: EDebtStatus,
    userTypeInfo: BadgeInfoDto,
    levelId: number,
    levelInfo: BadgeInfoDto,
    branchId: number,
    branchInfo: BadgeInfoDto,
    avatar: string,
    employeeId: number,
    interestRate: number,
    money: number,
    startDate: string,
    endDate: string,
    note: string,
    paymentType: EPaymentType,
    debtStatus: EDebtStatus
    interest: number
    totalPaid: number,
    creationTime: string,
    creatorUser:string,
    updatedUser:string,
    updatedTime:string,
}

export enum EPaymentType {
    All = 0,
    Salary = 1,
    Cash = 2
}

export enum EDebtStatus {
    Inprogress = 1,
    Done = 2,
}

export enum EUserStatus{
    DangLam,
    Tamnghi,
    Danghi,
    Nghisinh
}

export interface DebtInputFilterDto extends GetInputFilterDto{
    debtstatusIds: number[],
    paymentTypeIds: number[]
}